noir-projects/aztec-nr/aztec/src/utils/bytes.nr

```diff
@@ -1,88 +1,307 @@
-// Converts the input bytes into an array of fields. A Field is ~254 bits meaning that each field can store 31 bytes.
+use crate::oracle::random::random;
+
+// Converts the input bytes into an array of fields. A Field is ~254 bits meaning that each field can store 31 whole bytes.
 //
-// Each 31 byte chunk is converted into a Field as if the chunk was the Field's big endian representation. If the last chunk
-// is less than 31 bytes long, then only the relevant bytes are conisdered.
+// The input bytes are chunked into chunks of 31 bytes. Each 31-byte chunk is viewed as big-endian, and is converted into a Field. If the last chunk
+// is less than 31 bytes long, then only the relevant bytes are considered.
 // For example, [1, 10, 3] is encoded as [1 * 256^2 + 10 * 256 + 3]
-pub fn bytes_to_fields<let N: u32>(input: [u8; N]) -> [Field; (N + 30) / 31] {
-    let mut dst = [0; (N + 30) / 31];
+// Note: ceil(N / 31) = (N + 30) / 31
+pub fn be_bytes_31_to_fields<let N: u32>(bytes: [u8; N]) -> [Field; (N + 30) / 31] {
```

Comment:
Optimised vs the previous implementation that did a load of `<` checks on indices at runtime.

Reply:
Did you measure this?

Reply:
Also no need to throw in optimizations on top of further changes :upside_down_face: this could just be its own thing

Reply:
> Did you measure this?

No, but each comparison is something like 15 constraints, so the `if byte_index < N` for every byte of the old version will have blown up.

Reply:
Yeah, this is an example of me seeing something and going "I should fix this" instead of "I should open an issue for this and come back to it".